### ISSUE TITLE
WebVTT should expose an inline-block backdrop element for cues

### ISSUE THREAD

**Issue by jernoble** [w3c/webvtt#516]
Safari and WebKit on macOS and iOS have a long-standing default treatment of text track cues that incorporates an `inline-block` wrapper around the cues, so as to provide the same rendering of cues as is available to native applications using system media frameworks. The styles for this `inline-block` wrapper are configurable through system defaults, allowing the user to choose between an `inline-block` style highlight, and a `inline` highlight. (See attached images for examples.)

Because this block is not exposed through WebVTT, it is very difficult (if not impossible) for inline- or page-provided styles to override this rendering.

<img width="783" alt="inline-block" src="https://github.com/w3c/webvtt/assets/1557428/355ca664-9f31-4de0-8a5a-4dcd2589620e">
<img width="783" alt="inline" src="https://github.com/w3c/webvtt/assets/1557428/3c90e1ca-d653-4073-bf55-93943833561e">

**Comment by jernoble** [comment]
Strawman solution 1: Add an additional wrapper anonymous block with a `display: inline-block` style, and add a specific selector for `::cue(selector)` that applies to the block.

Strawman solution 2: Allow the `display:` property to apply to the `::cue` and `::cue(selector)` pseudo element, with the allowed values of `inline` and `inline-block`.

**Comment by jernoble** [comment]
As an internal implementation detail, WebKit uses the pseudo-element `::-webkit-media-text-track-display-backdrop` to refer to the `inline-block` element which wraps each `::cue`. While that's an unreasonably verbose pesudo name, something like `::cue-backdrop` would be sufficiently succinct. The `::cue-backdrop` pseudo-element could be restricted to a short list of CSS properties; such as `background`, `border-radius`, and `padding`.

**Comment by jernoble** [comment]
No idea how this maps to TTML. For HTML, it's definitely equivalent to a `div` insofar as a `p` is just a `div` with style.

**Comment by jernoble** [comment]
Pinging this thread. Strawman proposal would be that the "window" of a WebVTT cue should be selectable by a `::cue-window` selector.

**Comment by nigelmegitt** [comment]
@jernoble can you define "window"?

**Comment by jernoble** [comment]
What definition are you looking for? CEA-708 already defines "caption windows", as well as defines display properties of those windows, including background color, border effects, and transparency. But in the context of WebVTT, I would define a "cue window" as an "inline-block element wrapping the WebVTT Internal Objects generated by the cue".

**Comment by jernoble** [comment]
Ping on this issue. The ability for web pages to override system-provided styling of "caption windows" is critical for interoperability and use of WebVTT in web pages generally.

**Comment by silviapfeiffer** [comment]
I like the idea of ::cue-window for dealing with the background. Can I suggest preparing a pull request? That makes the life of the current maintainer a bit easier...

**Comment by gkatsev** [comment]
I would be in support of exposing `::cue-window`. It would basically be an alias for the existing `::-webkit-media-text-track-display-backdrop` element in safari?

It would be nice to figure out the TTML mapping, but that can be done separately.

**Comment by jernoble** [comment]
I posted a straw man pull request for this issue as PR# 528. I ran this idea past some of our CSS experts internally, and they balked at the term "window", and suggested the term "container" instead. They also suggested we disambiguate between the "cue inline background box" and the "cue block background box", so I updated those definitions as part of this PR.

**Comment by silviapfeiffer** [comment]
Has this been implemented by any other Web browser? IF so, then we should
align and not invent new ways of calling things.

If not, then I'm all for following an engineer's suggestions who is
currently in the process of implementing this!

Echo of a voice from the past, FWIW - feel free to ignore. :-)

**Comment by jernoble** [comment]
As far as I know, Blink and WebKit both have this concept, and both call it a "backdrop" internally. I'm not aware of any other browser implementing a public name for this, however.

I'm super happy to bike shed the CSS selector name itself; like I said, this is a straw man proposal.

**Comment by nigelmegitt** [comment]
Am I right in understanding that this falls somewhere between a (WebVTT region object)[https://www.w3.org/TR/webvtt1/#webvtt-region-object] and a [WebVTT cue background box](https://www.w3.org/TR/webvtt1/#webvtt-cue-background-box)?

I'm unclear why the region object isn't sufficient already.

**Comment by jernoble** [comment]
> Am I right in understanding that this falls somewhere between a [WebVTT region object](https://www.w3.org/TR/webvtt1/#webvtt-region-object) and a [WebVTT cue background box](https://www.w3.org/TR/webvtt1/#webvtt-cue-background-box)?

Yes; it is a block-level object (like region) which wraps an inline level object (the cue _inline_ background box).

> I'm unclear why the region object isn't sufficient already.

I originally tried to wrap cues without regions in an "anonymous" region, but it resulted in an _extremely invasive_ change to the region algorithms. Normally, cues within regions are laid out depending on the region's location, but defining an "anonymous" region required having the region is laid out depending on its cue's location. Making WebVTT regions be capable of both layout modes effectively meant creating something that was named similarly but had entirely different behavior.

Simpler being better, I decided to instead define a block-level container for each cue. In addition to being simpler and requiring zero changes to the [Rendering] (https://www.w3.org/TR/webvtt1/#rendering) algorithms, this is already implemented by two browser engines (Blink and WebKit).

**Comment by nigelmegitt** [comment]
>Yes; it is a block-level object (like region) which wraps an inline level object (the cue inline background box).


In https://github.com/w3c/webvtt/pull/528#issuecomment-2729013292 I just proposed `::cue-block` on the basis of this comment.
